pygeobase/io_base.py

```diff
@@ -312,11 +443,55 @@ def _open(self, gpi):
         cell = self.grid.gpi2cell(gpi)
         filename = os.path.join(self.path, self.fn_format.format(cell))
 
-        if self.previous_cell != cell:
-            self.close()
+        if self.mode == 'r':
+            if self._read_bulk:
```

Comment:
Is there really a use case in which we would create a new ioclass object for every grid point?